### DISCUSSION TITLE
Running shell commands with piping operator "|" results in unknown operator error

### DISCUSSION THREAD

**Discussion by adam2392** [nat-n/poethepoet#94] [Q&A] [answered]
Hi,

Great tool!

I'm trying to run the following command inside my `pyproject.toml` file with poetry.

```
[tool.poe.tasks]
clean_so = """
find . -name "*.so" | xargs rm -f
find . -name "*.pyd" | xargs rm -f
"""
```

However, when I run `poetry run poe clean_so`, I get the following error: `find: |: unknown primary or operator`. Is there a way to run shell commands with shell-command operators?

**Comment by nat-n** [comment, marked as answer]
Indeed, the default task type (cmd) is executed as a subprocess without a shell. If you want to use shell features like the pipe operator then you need to create a shell task which can be done like so:

```toml
[tool.poe.tasks]
clean_so.shell = """
find . -name "*.so" | xargs rm -f
find . -name "*.pyd" | xargs rm -f
"""
```

Be aware that this will only work as expected if a POSIX shell can be found in the host environment, which is not always the case on windows.

See the [section of the readme about task types](https://github.com/nat-n/poethepoet#types-of-task) for more details.

**Reply by adam2392** [reply]
Thanks!
